lms/djangoapps/verify_student/views.py

```diff
@@ -368,7 +368,8 @@ def get(
 
         # Remember whether the user is upgrading
         # so we can fire an analytics event upon payment.
-        request.session['attempting_upgrade'] = (message == self.UPGRADE_MSG)
+        if message == self.UPGRADE_MSG:
+            request.session['attempting_upgrade'] = True
```

Comment:
What happens if a user enters the upgrade flow, but doesn't complete payment?  If the user then goes to purchase another course (not upgrade), then an "upgrade" event would be fired, even though the user isn't upgrading.